jdaviz/configs/imviz/plugins/subset_control/subset_control.py

```diff
@@ -0,0 +1,102 @@
+from glue.core.roi import CircularROI, EllipticalROI, RectangularROI
+from glue.core.subset import Subset
+from glue.core.message import SubsetUpdateMessage, SubsetDeleteMessage
+from traitlets import Float, List
+
+from jdaviz.core.events import SnackbarMessage
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin
+from jdaviz.utils import load_template
+
+__all__ = ['SubsetControl']
+
+
+@tray_registry('imviz-subset-control', label="Subset Control")
+class SubsetControl(TemplateMixin):
+    template = load_template("subset_control.vue", __file__).tag(sync=True)
+
+    subset_items = List([]).tag(sync=True)
+    new_subset_angle = Float(default=0).tag(sync=True)
+    new_subset_x = Float().tag(sync=True)
+    new_subset_y = Float().tag(sync=True)
+
+    # TODO:
+    # 1. Let user select which subset to operate on.
+    # 2. Hide angle field when it does not make sense (e.g., circle)
+    # 3. Grab new values from user inputs
+    # 4. Apply new values to Subset
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._selected_subset_label = ''
+        self._all_subsets = {}
+
+        # NOTE: Subscribing to SubsetCreateMessage breaks Subset interactive creation.
+        self.hub.subscribe(self, SubsetUpdateMessage,
+                           handler=lambda x: self._on_viewer_subset_changed())
+        self.hub.subscribe(self, SubsetDeleteMessage,
+                           handler=lambda x: self._on_viewer_subset_changed())
+
+    def get_subset_rois(self):
+        """Get a list of all current Subset ROI objects."""
+        # Even though Subset is global, still need to access it from viewer.
+        viewer = self.app.get_viewer('viewer-1')
+        rois = {}
+        for layer_state in viewer.state.layers:
+            if hasattr(layer_state, 'layer') and isinstance(layer_state.layer, Subset):
+                rois[layer_state.layer.label] = layer_state.layer.subset_state.roi
+        return rois
+
+    def _on_viewer_subset_changed(self):
+        """Callback method for when a subset is updated or deleted."""
+        self._all_subsets = self.get_subset_rois()
+        self.subset_items = sorted(self._all_subsets.keys())
+
+    def _has_subset_angle(self):
+        cur_roi = self._all_subsets[self._selected_subset_label]
+        has_angle = False
+        if isinstance(cur_roi, (EllipticalROI, RectangularROI)):
+            has_angle = True
+        return has_angle
+
+    # TODO: How to make v-show="has_subset_angle" work?
```

Comment:
@mariobuikhuizen , any advise on how to hook up `v-show` here? My goal is to hide "Angle" text field when this function evaluates to `False`.

Reply:
@pllim, this has to be done with a traitlet, a method call does not return anything to the frontend. The traitlet could be updated from this method.

Reply:
Ah, that makes sense. I only found JS solutions in SO and got confused. Thanks!